p2p: rename GetAddresses -> GetAddressesUnsafe

danielabrozzoni opened the pull request:
Rename GetAddresses to GetAddressesUnsafe to make it clearer that this function should only be used in trusted contexts. This helps avoid accidental privacy leaks by preventing the uncached version from being used in non-trusted scenarios, like P2P.

Additionally, better reflect in the documentation that the two methods should be used in different contexts.
Also update the outdated "call the function without a parameter" phrasing in the cached version. This wording was accurate when the cache was introduced in #18991, but became outdated after later commits (f26502e9fc8a669b30717525597e3f468eaecf79, 81b00f87800f40cb14f2131ff27668bd2bb9e551) added parameters to each
function, and the previous commit changed the function naming completely.

stickies-v reviewed (commented):
Concept ACK. I would strongly prefer to go further and rename the trusted function to `GetAddressesUncached`. Overloads that behave vastly differently and can lead to dangerous behaviour is a big red flag for me.

brunoerg commented:
I agree with @stickies-v. It's weird to have two functions with the same name but with a relevant behavior difference, renaming one of the functions seems appropriate. In any case, Concept ACK about improving the documentation.

danielabrozzoni commented:
@stickies-v @brunoerg thank you for the suggestion!

I agree, I think renaming makes the code clearer and reduces the chances of accidentally using the uncached version and leaking addrman addresses :)

I updated the PR title/description and pushed a commit to rename GetAddresses to GetAddressesUncached!

l0rinc commented on src/net.h:
```diff
@@ -1170,18 +1170,22 @@ class CConnman
     // Addrman functions
     /**
      * Return all or many randomly selected addresses, optionally by network.
+     * This function does not use an address response cache and should only be
```
Even though the PR is quite simple, I'd separate the comment-changing-commits from the refactoring commits. I guess it may not make a lot of sense doing the rename with a scripted diff since it might confuse the two similarly named methods.

stickies-v replied:
nit: documentation could be made more consistent, as per e.g. the below diff

<details>
<summary>git diff on df65b527f0</summary>

```diff
diff --git a/src/net.h b/src/net.h
index 61b25bd47d..eb08d9de05 100644
--- a/src/net.h
+++ b/src/net.h
@@ -1169,9 +1169,10 @@ public:
 
     // Addrman functions
     /**
-     * Return all or many randomly selected addresses, optionally by network.
-     * This function does not use an address response cache and should only be
-     * used in trusted contexts.
+     * Return randomly selected addresses. This function does not use the address response cache and
+     * should only be used in trusted contexts.
+     *
+     * An untrusted caller (e.g. from p2p) should instead use @ref GetAddresses to use the cache.
      *
      * @param[in] max_addresses  Maximum number of addresses to return (0 = all).
      * @param[in] max_pct        Maximum percentage of addresses to return (0 = all). Value must be from 0 to 100.
@@ -1180,12 +1181,18 @@ public:
      */
     std::vector<CAddress> GetAddressesUncached(size_t max_addresses, size_t max_pct, std::optional<Network> network, const bool filtered = true) const;
     /**
-     * Return all or many addresses, using the address response cache.
-     * Cache is used to minimize topology leaks, so it should be used for all
-     * non-trusted calls, for example, p2p.
+     * Return addresses from the per-requestor cache. If no cache entry exists, it is populated with
+     * randomly selected addresses. This function can be used in untrusted contexts.
+     *
+     * A trusted caller (e.g. from RPC or a peer with addr permission) can use
+     * @ref GetAddressesUncached to avoid using the cache.
      *
-     * A non-malicious call (from RPC or a peer with addr permission) should
-     * instead use GetAddressesUncached to avoid using the cache.
+     * @param[in] requestor      The requesting peer. Used to key the cache to prevent privacy leaks.
+     * @param[in] max_addresses  Maximum number of addresses to return (0 = all). Ignored when cache
+     *                           already contains an entry for requestor.
+     * @param[in] max_pct        Maximum percentage of addresses to return (0 = all). Value must be
+     *                           from 0 to 100. Ignored when cache already contains an entry for
+     *                           requestor.
      */
     std::vector<CAddress> GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct);
 

```
</details>

l0rinc reviewed (approved):
Concept ACK, please consider splitting different change types to separate commits and adding extra qualifications to the methods that does the extra work.

danielabrozzoni commented on src/net.cpp:
```diff
@@ -3496,7 +3496,7 @@ CConnman::~CConnman()
 }
 
-std::vector<CAddress> CConnman::GetAddresses(size_t max_addresses, size_t max_pct, std::optional<Network> network, const bool filtered) const
+std::vector<CAddress> CConnman::GetAddressesUncached(size_t max_addresses, size_t max_pct, std::optional<Network> network, const bool filtered) const
```
I also don't love the `GetAddresses`/`GetAddressesUncached` naming either, but I agree with stickies that it's important that the unsafe version stands out.

What if we instead renamed them to `GetAddressesCached` / `GetAddressesUncached`? That way, there’s no ambiguity about what each function does, and the naming feels a bit more consistent.

danielabrozzoni committed:
p2p: rename GetAddresses -> GetAddressesUnsafe

Rename GetAddresses to GetAddressesUnsafe to make it clearer that this
function should only be used in trusted contexts. This helps avoid
accidental privacy leaks by preventing the uncached version from being
used in non-trusted scenarios, like P2P.

danielabrozzoni committed:
doc: clarify the GetAddresses/GetAddressesUnsafe documentation

Better reflect in the documentation that the two methods should be
used in different contexts.
Also update the outdated "call the function without a parameter" phrasing
in the cached version. This wording was accurate when the cache was
introduced in #18991, but became outdated after later commits
(f26502e9fc8a669b30717525597e3f468eaecf79,
81b00f87800f40cb14f2131ff27668bd2bb9e551) added parameters to each
function, and the previous commit changed the function naming completely.

Co-Authored-By: stickies-v <stickies-v@protonmail.com>

stickies-v reviewed (approved):
ACK df65b527f00cbf9fd6675bf22bb1285bfce24f66

danielabrozzoni reviewed (commented):
last push (1cb23997033c395d9ecd7bf2f54787b134485f41):
- rename `GetAddressesUncached` -> `GetAddressesUnsafe`
- improve documentation

l0rinc commented:
ACK 1cb23997033c395d9ecd7bf2f54787b134485f41

brunoerg reviewed (approved):
ACK 1cb23997033c395d9ecd7bf2f54787b134485f41

`GetAddresses{Unsafe}` sounds good and it's better than simply just changing the documentation.

luisschwab commented:
ACK 1cb23997033c395d9ecd7bf2f54787b134485f41

theStack reviewed (approved):
Code-review ACK 1cb23997033c395d9ecd7bf2f54787b134485f41

Nice documentation improvements, happy to see the method rename for improved clarity (fwiw, I'm generally not a huge fan of function overloading and think we tend to overuse it; it's often just confusing and makes it harder to find call-sites for a specific function, especially across multiple commits, e.g. for investigating past changes via `git log -S`).

mzumsande reviewed (commented):
Code review ACK 1cb23997033c395d9ecd7bf2f54787b134485f41
